pinecone/core/utils/__init__.py

```diff
@@ -109,3 +109,10 @@ def load_strings_public(proto_arr: 'vector_column_service_pb2.NdArray') -> List[
 def warn_deprecated(description: str = '', deprecated_in: str = None, removal_in: str = None):
     message = f'DEPRECATED since v{deprecated_in} [Will be removed in v{removal_in}]: {description}'
     warnings.warn(message, FutureWarning)
+
+def check_kwargs(caller, given):
+    import inspect
```

Comment:
I don't think it's common practice to have an import statement inside of a function definition. Seems like this could have some performance implication if we start using `check_kwargs` in more places, particularly on data plane operations which may be high volume. I would move this import to the top of the file.

And I know there's a ton of untested code in this project, but it would be great to see a test case or two showing what this does in different scenarios.